ref/haskell/src/Codec/Binary/Bech32.hs

```diff
@@ -97,30 +97,32 @@ bech32Decode bech32 = do
     guard $ bech32VerifyChecksum hrp' dat'
     return (hrp', take (BS.length dat - 6) dat')
 
-toBase32 :: [Word8] -> [Word5]
-toBase32 dat = pad ++ result
+convertBits :: [Word] -> Int -> Int -> Bool -> Maybe [Word]
```

Comment:
I'd like a little documentation about covertBits as a comment:
* frombits and twobits must be positive and less than the bitSize of Word.
* Every value of Word must be strictly less than 2^frombits.
* The conversion is Big Endien.